lib/active_admin/filters/resource_extension.rb

```diff
@@ -107,7 +107,11 @@ def filter_lookup
 
       # @return [Array] The array of default filters for this resource
       def default_filters
-        default_association_filters + default_content_filters + custom_ransack_filters
+        [].tap do |result|
+          result.concat default_association_filters if namespace.include_default_association_filters
+          result.concat default_content_filters
+          result.concat custom_ransack_filters
+        end
```

Comment:
`tap` isn't beginner-friendly, because it isn't clear what it does. this would be cleaner:
```ruby
filters = []
filters << default_association_filters if namespace.include_default_association_filters
filters << default_content_filters
filters << custom_ransack_filters
filters